cluster-state-service/internal/features/e2etasksteps/common_steps.go

```diff
@@ -179,3 +220,111 @@ func startNTasks(numTasks int, startedBy string, ecsWrapper wrappers.ECSWrapper)
 		EcsTaskList = append(EcsTaskList, ecsTask)
 	}
 }
+
+func createInstanceProfile(iamWrapper wrappers.IAMWrapper) error {
+	assumeRolePolicy := `{
+		"Version": "2012-10-17",
+		"Statement": [
+		{
+		"Effect": "Allow",
+		"Principal": {
+			"Service": "ec2.amazonaws.com"
+		},
+		"Action": "sts:AssumeRole"
+		}
+		]
+	}`
+
+	err := iamWrapper.GetRole(&roleName)
+	if err != nil {
+		if awsErr, ok := errors.Cause(err).(awserr.Error); ok && awsErr.Code() == "NoSuchEntity" {
+			err = iamWrapper.CreateRole(&roleName, &assumeRolePolicy)
+			if err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+
+	err = iamWrapper.GetInstanceProfile(&instanceProfileName)
+	if err != nil {
+		if awsErr, ok := errors.Cause(err).(awserr.Error); ok && awsErr.Code() == "NoSuchEntity" {
+			err = iamWrapper.CreateInstanceProfile(&instanceProfileName)
+			if err != nil {
+				return err
+			}
+		} else {
+			return err
+		}
+	}
+
+	err = iamWrapper.AttachRolePolicy(&policyARN, &roleName)
+	if err != nil {
+		return err
+	}
+
+	err = iamWrapper.AddRoleToInstanceProfile(&roleName, &instanceProfileName)
+	if err != nil {
+		if awsErr, ok := errors.Cause(err).(awserr.Error); ok && awsErr.Code() != "EntityAlreadyExists" && awsErr.Code() != "LimitExceeded" {
```

Comment:
When IAM role and profile already existed and is tied to each others. In that case, we can just move on.